src/tile.cpp

```diff
@@ -933,7 +933,7 @@ void Tile::addThing(int32_t, Thing* thing)
 
 			items = makeItemList();
 			items->insert(items->getBeginDownItem(), item);
-			++items->downItemCount;
+			items->addDownItemCount(1);
```

Comment:
downItemCount is a private member variable. It was accessible before because Tile was a friend of TileItemVector(which breaks Encapsulation principle), but that's not the biggest problem. The biggest issue is safety - a friend class can improperly delete the TileItemVector (by deleting a pointer to a vector), so to prevent such mistakes I've decided to remove the friend relation.